Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,540 @@
+
+@[simp]
+theorem coe_toContinuousMap (f : C_c(α, β)) : (f.toContinuousMap : α → β) = f :=
+  rfl
+
+@[ext]
+theorem ext {f g : C_c(α, β)} (h : ∀ x, f x = g x) : f = g :=
+  DFunLike.ext _ _ h
+
+@[simp]
+theorem coe_mk (f : C(α, β)) (h : HasCompactSupport f) : ⇑(⟨f, h⟩ : C_c(α, β)) = f :=
+  rfl
+
+/-- Copy of a `CompactlySupportedContinuousMap` with a new `toFun` equal to the old one. Useful
+to fix definitional equalities. -/
+protected def copy (f : C_c(α, β)) (f' : α → β) (h : f' = f) : C_c(α, β) where
+  toFun := f'
+  continuous_toFun := by
+    rw [h]
+    exact f.continuous_toFun
+  hasCompactSupport' := by
+    simp_rw [h]
+    exact f.hasCompactSupport'
+
+@[simp]
+theorem coe_copy (f : C_c(α, β)) (f' : α → β) (h : f' = f) : ⇑(f.copy f' h) = f' :=
+  rfl
+
+theorem copy_eq (f : C_c(α, β)) (f' : α → β) (h : f' = f) : f.copy f' h = f :=
+  DFunLike.ext' h
+
+theorem eq_of_empty [IsEmpty α] (f g : C_c(α, β)) : f = g :=
+  ext <| IsEmpty.elim ‹_›
+
+/-- A continuous function on a compact space has automatically compact support. -/
+@[simps]
+def ContinuousMap.liftCompactlySupported [CompactSpace α] : C(α, β) ≃ C_c(α, β) where
+  toFun f :=
+    { toFun := f
+      continuous_toFun := f.continuous
+      hasCompactSupport' := HasCompactSupport.of_compactSpace f }
+  invFun f := f
+  left_inv _ := rfl
+  right_inv _ := rfl
+
+end Basics
+
+/-! ### Algebraic structure
+
+Whenever `β` has the structure of continuous additive monoid and a compatible topological structure,
+then `C_c(α, β)` inherits a corresponding algebraic structure. The primary exception to this is that
+`C_c(α, β)` will not have a multiplicative identity.
+-/
+
+section AlgebraicStructure
+
+variable [TopologicalSpace β] (x : α)
+
+instance [Zero β] : Zero C_c(α, β) where
+  zero := { toFun := (0 : C(α, β))
+            continuous_toFun := (0 : C(α, β)).2
+            hasCompactSupport' := by simp [HasCompactSupport, tsupport] }
+
+instance [Zero β] : Inhabited C_c(α, β) :=
+  ⟨0⟩
+
+@[simp]
+theorem coe_zero [Zero β] : ⇑(0 : C_c(α, β)) = 0 :=
+  rfl
+
+theorem zero_apply [Zero β] : (0 : C_c(α, β)) x = 0 :=
+  rfl
+
+instance [MulZeroClass β] [ContinuousMul β] : Mul C_c(α, β) :=
+  ⟨fun f g => ⟨f * g, HasCompactSupport.mul_left g.2⟩⟩
+
+@[simp]
+theorem coe_mul [MulZeroClass β] [ContinuousMul β] (f g : C_c(α, β)) : ⇑(f * g) = f * g :=
+  rfl
+
+theorem mul_apply [MulZeroClass β] [ContinuousMul β] (f g : C_c(α, β)) : (f * g) x = f x * g x :=
+  rfl
+
+/-- the product of `f : F` assuming `ContinuousMapClass F α γ` and `ContinuousSMul γ β` and
+`g : C_c(α, β)` is in `C_c(α, β)` -/
+instance [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
+    {F : Type*} [FunLike F α γ] [ContinuousMapClass F α γ] : SMul F C_c(α, β) where
+  smul f g :=
+    ⟨⟨fun x ↦ f x • g x, (map_continuous f).smul g.continuous⟩, g.hasCompactSupport'.smul_left⟩
+
+@[simp]
+theorem coe_smulc [Zero β] [TopologicalSpace γ] [SMulZeroClass γ β] [ContinuousSMul γ β]
```

Comment:
Can you explain why this is called `coe_smulc`? I couldn't figure it out. What does the `c` mean? Probably adding a docstring about it would be nice. Sorry if this was addressed in earlier reviews; I didn't read them all.

Reply:
"c" is for Continuous. I thought it would be nice to remember that `SMul` is defined for `ContinuousMapClass F α γ`, not just `Funlike F α γ`. But if you don't think it is necessary, I will remove it.